src/univariate/affine.jl

```diff
@@ -0,0 +1,332 @@
+"""
+    AffineDistribution(μ, σ, ρ)
+
+A shifted and scaled (affinely transformed) version of `ρ`.
+
+If ``Z`` is a random variable with distribution `ρ`, then `AffineDistribution(μ, σ, ρ)` is the
+distribution of the random variable
+```math
+X = μ + σ * Z
+```
+
+If `ρ` is a discrete univariate distribution, the probability mass function of the transformed 
+distribution is given by
+```math
+P(X = x) = P\\left(Z = \\frac{x-μ}{σ} \\right).
+```
+
+If `ρ` is a continuous univariate distribution with probability density function `f_Z`,
+the probability density function of the transformed distribution is given by
+```math
+f_X(x) = \\frac{1}{|σ|} f_Z\\left( \\frac{x-μ}{σ} \\right).
+```
+
+Generally, it is recommended to not use the `AffineDistribution` constructor but
+`+`, `-`, `*`, and `/` to construct distributions of affine transformations. The latter fall back
+to constructing an `AffineDistribution` but can return more optimized distributions, e.g.,
+if `ρ` is from a location-scale family.
+
+```julia
+d = σ * ρ + μ       # Create location-scale transformed distribution
+params(d)           # Get the parameters, i.e. (μ, σ, ρ)
+```
+"""
+struct AffineDistribution{N,S<:ValueSupport,Tμ,Tσ,D<:Distribution{ArrayLikeVariate{N},S}} <: Distribution{ArrayLikeVariate{N},S}
+    μ::Tμ
+    σ::Tσ
+    dist::D
+end
+
+
+#### Aliases
+@deprecate LocationScale(args...; kwargs...) AffineDistribution(args...)
+const ContinuousAffine{F,Tμ,Tσ,D} = AffineDistribution{F,Continuous,Tμ,Tσ,D}
+const DiscreteAffine{F,Tμ,Tσ,D} = AffineDistribution{F,Discrete,Tμ,Tσ,D}
+
+const UnivariateAffine{S,Tμ,Tσ,D} = AffineDistribution{Univariate,S,Tμ,Tσ,D}
+const MultivariateAffine{S,Tμ,Tσ,D} = AffineDistribution{Multivariate,S,Tμ,Tσ,D}
+
+
+#### Constructors
+
+"""
+    +(μ, dist)
+
+Return a version of `dist` that has been translated by `μ`.
+"""
+Base.:+(μ::Real, ρ::UnivariateDistribution) = AffineDistribution(μ, one(μ), ρ)
+
+function Base.:+(μ::AbstractArray{<:Real, N}, ρ::Distribution{ArrayLikeVariate{N}, S}) where {N, S}
+    if size(μ) ≠ size(dist) 
+        throw(DimensionMismatch("array and distribution have different sizes."))
+    end
+    return AffineDistribution(μ, Diagonal(Ones(μ)), ρ)
+end
+
+
+"""
+    *(σ, dist)
+
+Return a version of `dist` that has been scaled by `σ`, where `σ` can be any real number or 
+matrix.
+"""
+function Base.:*(σ::Real, ρ::UnivariateDistribution)
+    if iszero(σ)
+        throw(ArgumentError("scale must be non-zero"))
+    end
+    return AffineDistribution(zero(σ), σ, ρ)
+end
+
+function Base.:*(σ::Real, ρ::Distribution)
+    if iszero(σ)
+        throw(ArgumentError("scale must be non-zero"))
+    end
+    return AffineDistribution(Zeros{eltype(ρ)}(size(ρ)), σ, ρ)
```

Comment:
I think we should be consistent and base the type on `σ`:
 ```suggestion
    return AffineDistribution(Zeros{typeof(σ)}(size(ρ)), σ, ρ)
```